extensions/infinispan-embedded/runtime/pom.xml

```diff
@@ -30,6 +30,11 @@
             <groupId>io.smallrye</groupId>
             <artifactId>smallrye-metrics</artifactId>
         </dependency>
+        <!-- Hack to get it to have the API Infinispan uses available -->
```

Comment:
Could you elaborate why do you call this a hack @wburns ?

Reply:
I'd rather ask why aren't we importing the `quarkus-smallrye-metrics` module here instead?

Reply:
> Could you elaborate why do you call this a hack @wburns ?

It isn't anymore, I forgot to remove the comment is all.

> I'd rather ask why aren't we importing the `quarkus-smallrye-metrics` module here instead?

But I can try this instead and see if it still works. Either way I can make a simple PR here in just a bit.

Reply:
For me, it depends if you need the API or an implementation.

If you just need the API, I wouldn't make the implementation mandatory.

@jmartisk WDYT?

Reply:
AFAIK the SmallRye Metrics implementation is mandatory for Infinispan to be able to run (ideally it should be able to run without metrics dependency at all if they are disabled, but that's not the case right now)

Reply:
I created https://github.com/quarkusio/quarkus/pull/7935 to remove the comment. Let me know if I should change the dependency though.

Reply:
Other extensions that make use of Metrics declare the `quarkus-smallrye-metrics` dependency in their runtime module, so I'd vote for yes

Reply:
Ah ah, I missed the `smallrye-metrics` dependency just above. So yes, we need the extension to be used.

Reply:
> right now

Now (10.1.5.final) it should be able to work without Smallrye metrics being present in classpath at runtime. It's truly optional.